langgraph/checkpoint/memory.py

```diff
@@ -1,47 +1,84 @@
+import asyncio
+import pickle
 from collections import defaultdict
-from typing import Optional
+from typing import AsyncIterator, Iterator, Optional
 
-from langchain_core.pydantic_v1 import Field
 from langchain_core.runnables import RunnableConfig
 
-from langgraph.checkpoint.base import BaseCheckpointSaver, Checkpoint, CheckpointTuple
+from langgraph.checkpoint.base import (
+    BaseCheckpointSaver,
+    Checkpoint,
+    CheckpointAt,
+    CheckpointTuple,
+    SerializerProtocol,
+)
 
 
 class MemorySaver(BaseCheckpointSaver):
-    storage: defaultdict[str, dict[str, Checkpoint]] = Field(
-        default_factory=lambda: defaultdict(dict)
-    )
+    serde = pickle
 
-    def get(self, config: RunnableConfig) -> Optional[Checkpoint]:
-        if value := self.get_tuple(config):
-            return value.checkpoint
+    storage: defaultdict[str, dict[str, Checkpoint]]
+
+    def __init__(
+        self,
+        *,
+        serde: Optional[SerializerProtocol] = None,
+        at: Optional[CheckpointAt] = None,
+    ) -> None:
```

Comment:
Could we add docstrings :)

Reply:
haha yea but only after settling on final checkpointer interface